### PR TITLE
Added Test Coverage to Int32 and Make Sure Tests Succeed

### PR DESCRIPTION
**Pull request by FrostyTheSouthernSnowman**
I added a bunch of tests in `test/test_dtype.py` for int32.

Some of the tests were failing due to typecasting issues. I made some changes to `tinygrad/runtime/ops_cpu.py` to fix those issues.

This PR solves #1170

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/runtime/ops_cpu.py                8               5
------------------------------------------------------------
lines added in the tinygrad folder: 3
```

**Comment by geohot** [comment]
Tests failing.

**Comment by FrostyTheSouthernSnowman** [comment]
Yep. I am going to work on that. I think the `match_types` function in `tinygrad/runtime/ops_cpu.py` function doesn't play well with pytorch and seems to be getting rid of gradients on some Tensors. My plan is to unit test `match_types` and then work on debugging those issues.

**Comment by FrostyTheSouthernSnowman** [comment]
PR ready to be merged

File: tinygrad/runtime/ops_cpu.py [hunk 1]
```diff
@@ -26,16 +30,14 @@ def mulacc(a, b, new_shape):
...
-  BinaryOps.MAX: np.maximum, BinaryOps.CMPEQ: lambda x,y: (x==y).astype(np.promote_types(x.dtype,y.dtype)), BinaryOps.MUL: lambda x, y: np.multiply(*match_types(x, y)), UnaryOps.SQRT: np.sqrt,
+  BinaryOps.MAX: np.maximum, BinaryOps.CMPEQ: lambda x,y: (x==y).astype(np.promote_types(x.dtype,y.dtype)), BinaryOps.ADD: lambda x, y: np.add(*match_types(x, y)), 
+  BinaryOps.SUB: lambda x, y: np.subtract(*match_types(x, y)), BinaryOps.MUL: lambda x, y: np.multiply(*match_types(x, y)), 
+  BinaryOps.DIV: lambda x, y: np.divide(*match_types(x, y)), UnaryOps.SQRT: np.sqrt,
   MovementOps.PERMUTE: lambda x, order: x.transpose(order), MovementOps.PAD: np.pad, MovementOps.EXPAND: np.broadcast_to,
-  MovementOps.STRIDE: lambda x, arg: x[tuple(slice(None, None, i) for i in arg)],
-  FusedOps.MULACC: einsum_mulacc(lambda s,a,b: np.einsum(s, a.copy(), b.copy(), optimize=True), lambda x: x.strides, np.broadcast_to),
+  MovementOps.STRIDE: lambda x, arg: x[tuple(slice(None, None, i) for i in arg)], 
+  FusedOps.MULACC: einsum_mulacc(lambda s,a,b: np.einsum(s, *match_types(a.copy(), b.copy())), lambda x: x.strides, np.broadcast_to),
```
Comment (python273) [inline tinygrad/runtime/ops_cpu.py:40]: need to put back `optimize`

Comment (FrostyTheSouthernSnowman) [reply tinygrad/runtime/ops_cpu.py:40]: oh yeah! let me do that real quick!

Comment (FrostyTheSouthernSnowman) [reply tinygrad/runtime/ops_cpu.py]: thanks for reminding me

File: test/test_dtype.py [hunk 2]
```diff
@@ -3,7 +3,9 @@
 from tinygrad.helpers import getenv, DType, DEBUG
 from tinygrad.lazy import Device
 from tinygrad.tensor import Tensor, dtypes
+from tinygrad.runtime.ops_cpu import match_types
 from extra.utils import OSX
+import torch
```
Comment (geohot) [inline test/test_dtype.py:8]: Why?

File: tinygrad/runtime/ops_cpu.py [hunk 3]
```diff
@@ -16,6 +16,10 @@ def shape_to_axis(old_shape:Tuple[int, ...], new_shape:Tuple[int, ...]) -> Tuple
   MovementOps.RESHAPE: lambda x, arg: x.reshape(arg), MovementOps.SHRINK: lambda x, arg: x[tuple(slice(p[0], p[1], None) for p in arg)],
 }
 
+def match_types(x, y):
```
Comment (geohot) [inline tinygrad/runtime/ops_cpu.py:19]: This is making many needless copies.

"By default, astype always returns a newly allocated array"

Comment (FrostyTheSouthernSnowman) [reply tinygrad/runtime/ops_cpu.py:19]: Sorry. I changed it to use copy=False for both x and y

File: test/test_dtype.py (hunk 2 above)
Comment (FrostyTheSouthernSnowman) [reply test/test_dtype.py:8]: I had unit tests for match_types, but I removed them because they were breaking the CI in weird ways. I forgot to remove the imports while working on fixing the CI
